sips/sip-030/sip-030-wallet-interface.md

```diff
@@ -0,0 +1,571 @@
+- Specify compatible RPC methods and payloads for wallet interaction.
+- Define a recommended provider object and its discovery mechanism.
+
+> **Note**: The methods can be used via the mentioned browser object (i.e., via the `window.WalletProvider.request` method), but also through similar interfaces like WalletConnect.
+
+## Backwards Compatibility
+
+The implementation of this proposal is not necessarily backward compatible.
+Notably, app-specific private keys for domains have been removed from the connect protocol, but can still be accessed via the `gaiaAppKey` field in `stx_getAccounts`.
+
+Wallets implementing the new standard may maintain the previous system to support legacy applications during a transition period or indefinitely.
+Existing applications using the current Auth system should continue to operate, but immediate changes are recommended once this SIP is ratified.
+
+## Reference Implementations
+
+### Notes on Serialization
+
+To ensure serializability, consider these notes:
+
+- Enums are serialized as human-readable strings.
+- BigInts are serialized as numbers, strings, or anything that can be parsed by the JavaScript BigInt constructor.
+- Bytes are serialized as hex-encoded strings (without a 0x prefix).
+- Predefined formats from previous SIPs are used where applicable.
+- Addresses are serialized as Stacks c32-encoded strings.
+- Clarity values, post-conditions, and transactions are serialized to bytes (defined by SIP-005) and used as hex-encoded strings.
+
+### Methods
+
+This section defines the available methods, their parameters, and result structure.
+Parameters should be considered only as recommendations for the wallet, and the user/wallet may choose to ignore or override them.
+
+> Note: Optional params are marked with a `?`.
+
+Methods can be namespaced under `stx_` if used in more generic settings and other more Ethereum inspired domains.
+In other cases (e.g. `WalletConnect`), the namespace may already be given by metadata (e.g. a `chainId` field) and can be omitted.
+On the predominant `StacksProvider` global object, the methods can be used without a namespace, but wallets may add namespaced aliases for convenience.
+
+##### General parameters (for transaction methods)
+
+The following definitions can be used in the transaction methods.
+
+Parameter properties
+
+- `address?`: `string` address, Stacks c32-encoded, defaults to wallets current address
+- `network?`: `'mainnet' | 'testnet' | 'regtest' | 'devnet' | 'mocknet' | NetworkId`
+  - where `NetworkId extends string` and can be a network identifier, which already exists in the wallet (e.g. a URL or network name)
+- `fee?`: `number | string` BigInt constructor compatible value
+- `nonce?`: `number | string` BigInt constructor compatible value
+- `postConditions?`: `PostCondition[]`, defaults to `[]`
+  - where `PostCondition` is `string | object` hex-encoded or JSON representation
```

Comment:
Is there further structure to `PostCondition` you can put here, or link to?

Reply:
Great point added a note that this and ClarityValue are defined later in the document.